src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/logical.cpp

```diff
@@ -0,0 +1,182 @@
+#endif
+
+using namespace CPUTestUtils;
+using namespace ov::test::utils;
+
+namespace ov {
+namespace test {
+std::string LogicalLayerCPUTest::getTestCaseName(const testing::TestParamInfo<LogicalLayerCPUTestParamSet> &obj) {
+    const auto& [shapes, logicalType, secondInType, enforceSnippets] = obj.param;
+
+    std::ostringstream result;
+    result << "IS=(";
+    for (size_t i = 0lu; i < shapes.size(); i++) {
+        result << ov::test::utils::partialShape2str({shapes[i].first}) << (i < shapes.size() - 1lu ? "_" : "");
+    }
+    result << ")_TS=";
+    for (size_t i = 0lu; i < shapes.front().second.size(); i++) {
+        result << "{";
+        for (size_t j = 0lu; j < shapes.size(); j++) {
+            result << ov::test::utils::vec2str(shapes[j].second[i]) << (j < shapes.size() - 1lu ? "_" : "");
+        }
+        result << "}_";
+    }
+    result << "OpType=" << logicalType << "_";
+    if (ov::test::utils::LogicalTypes::LOGICAL_NOT != logicalType) {
+        result << "secondInType=" << secondInType << "_";
+    }
+    result << "_enforceSnippets=" << enforceSnippets;
+
+    return result.str();
+}
+
+void LogicalLayerCPUTest::SetUp() {
+    const auto& [shapes, logicalType, secondInType, enforceSnippets] = this->GetParam();
+    targetDevice = ov::test::utils::DEVICE_CPU;
+
+    const auto primitiveType = getPrimitiveType(logicalType);
+    const auto primiticePrcType = primitiveType == "ref" ? ov::element::f32.to_string() : "I8";
+    selectedType = primitiveType.empty() ? "" : primitiveType + "_" + primiticePrcType;
+
+    init_input_shapes(shapes);
+
+    if (enforceSnippets) {
+        configuration.insert(ov::intel_cpu::snippets_mode(ov::intel_cpu::SnippetsMode::IGNORE_CALLBACK));
+    } else {
+        configuration.insert(ov::intel_cpu::snippets_mode(ov::intel_cpu::SnippetsMode::DISABLE));
+    }
+
+    const auto prc = ov::element::boolean;  // Because ngraph supports only boolean input for logical ops
+    ov::ParameterVector params {std::make_shared<ov::op::v0::Parameter>(prc, inputDynamicShapes[0])};
+
+    std::shared_ptr<ov::Node> logical_node;
+    if (ov::test::utils::LogicalTypes::LOGICAL_NOT != logicalType) {
+        std::shared_ptr<ov::Node> secondInput;
+        if (ov::test::utils::InputLayerType::CONSTANT == secondInType) {
+            auto tensor = ov::test::utils::create_and_fill_tensor(prc, targetStaticShapes[0][1]);
+            secondInput = std::make_shared<ov::op::v0::Constant>(tensor);
+        } else {
+            auto param = std::make_shared<ov::op::v0::Parameter>(prc, inputDynamicShapes[1]);
+            secondInput = param;
+            params.push_back(param);
+        }
+        logical_node = ov::test::utils::make_logical(params[0], secondInput, logicalType);
+    } else {
+        logical_node = std::make_shared<ov::op::v1::LogicalNot>(params[0]);
+    }
+
+    logical_node->get_rt_info() = getCPUInfo();
+    function = std::make_shared<ov::Model>(logical_node, params, "Logical");
```

Comment:
NIT. Usually we joint these two actions and call `makeNgraphFunction` base class method.

Reply:
Thank you, done!